app/src/main/java/io/gnosis/safe/di/modules/ApplicationModule.kt

```diff
@@ -231,7 +235,24 @@ class ApplicationModule(private val application: Application) {
 
     @Provides
     @Singleton
-    fun providesKeyStorage(@ApplicationContext context: Context): KeyStorage = AndroidKeyStorage(context)
+    fun providesKeyStorage(@ApplicationContext context: Context): KeyStorage =
+        if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.M) {
+            val localeCopy = Locale.getDefault()
+            setLocale(context, Locale.ENGLISH)
+            val androidKeyStorage = AndroidKeyStorage(context)
+            setLocale(context, localeCopy)
+            androidKeyStorage
```

Comment:
Are we doing a quick locale swap here just for the key generation step?

Reply:
Yes. That is the solution I fond on the internet :-/

Reply:
this should be marked with //FIXME followed by a description of the reason for this workaround

Reply:
Added in fd5fbe51fb743768962bdba6212d18fe08007c17